client/state/sites/selectors.js

```diff
@@ -985,3 +985,19 @@ export function getCustomizerUrl( state, siteId ) {
 		'return': returnUrl
 	}, adminUrl );
 }
+
+/*
+ * Returns true if the site has unchanged site title
+ *
+ * @param {Object} state Global state tree
+ * @param {Object} siteId Site ID
+ * @return {Boolean} True if site title is default, false otherwise.
+ */
+export const hasDefaultSiteTitle = ( state, siteId ) => {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+	const slug = getSiteSlug( state, siteId );
+	return site.name === i18n.translate( 'Site Title' ) || slug.indexOf( site.name ) === 0;
```

Comment:
Oooooh! 💡 So you're checking whether the current title is what the slug starts with. Makes sense to me now to leave the logic as is, but I think we need to make this more self-explanatory. 

Worst case we add a comment, but I wonder if there's a way to make this more comprehensible with code changes only. 

But I think even `slug.startsWith( site.name )` doesn't really communicate this sufficiently. Maybe we really do need a comment here?